backend/schema_migration/migrate.py

```diff
@@ -1,88 +1,132 @@
-    s3_provider = S3Provider()
-    uri_provider = UriProvider()
+        CollectionQueryFilter()
+        published_collections = self.business_logic.get_collections(CollectionQueryFilter(is_published=True))
+        unpublished_collections = self.business_logic.get_collections(CollectionQueryFilter(is_published=False))
+
+        response = {"published": {}, "private": {}, "revision": {}}
+
+        def get_datasets(_version):
+            datasets = []
+            for dataset in _version.datasets:
+                datasets.append({"dataset_id": dataset.dataset_id.id, "dataset_version_id": dataset.version_id.id})
+            return datasets
+
+        collections = itertools.chain(unpublished_collections, published_collections)
+        # evaluate unpublished collections first, so that published versions are skipped if there is an active revision
+        has_revision = []  # list of collections to skip if published with an active revision
+        for collection in collections:
+            if collection.is_published() and collection.collection_id not in has_revision:
+                version = self.business_logic.get_collection_version(collection.version_id)
+                response["published"][version.collection_id.id] = get_datasets(version)
+            elif collection.is_unpublished_version():
+                has_revision.append(collection.collection_id)  # revision found, skip published version
+                version = self.business_logic.get_collection_version(collection.version_id)
+                response["revision"][version.version_id.id] = get_datasets(version)
+                # using version id instead of collection id
+            elif collection.is_initial_unpublished_version():
+                version = self.business_logic.get_collection_version(collection.version_id)
+                response["private"][version.collection_id.id] = get_datasets(version)
+        return response
+
+    def dataset_migrate(self, collection_id: str, dataset_id: str, dataset_version_id: str) -> Dict[str, str]:
+        raw_h5ad_uri = [
+            artifact.uri
+            for artifact in self.business_logic.get_dataset_artifacts(dataset_version_id)
+            if artifact.type == DatasetArtifactType.RAW_H5AD
+        ]
+        bucket_name, object_key = self.business_logic.s3_provider.parse_s3_uri(raw_h5ad_uri)
+        self.business_logic.s3_provider.download_file(bucket_name, object_key, "raw.h5ad")
+        cellxgene_schema.migrate("raw.h5ad", "migrated.h5ad", collection_id, dataset_id)
+        # TODO: define where this upload bucket actually should be
+        upload_bucket_name = os.environ["UPLOAD_BUCKET"]
+        dst_uri = f"{dataset_version_id}/migrated.h5ad"
+        self.business_logic.s3_provider.upload_file("migrated.h5ad", upload_bucket_name, dst_uri)
+        url = f"s3://{upload_bucket_name}/{dst_uri}"
+        new_dataset_version_id, _ = self.business_logic.ingest_dataset(collection_id, url, None, dataset_version_id)
```

Comment:
skip the step function trigger step, and return the parameters needs to trigger the step function in the response.